src/presenter/board-presenter.js

```diff
@@ -0,0 +1,105 @@
+import SiteFilters from '../view/site-menu-filters.js';
+import MainSort from '../view/main-sort.js';
+import TripList from '../view/trip-events-list.js';
+import PageHeader from '../view/page-header.js';
+import MainNavigation from '../view/menu-navigation.js';
+import TripCost from '../view/trip-info-cost.js';
+import InfoMain from '../view/trip-info-main.js';
+import PageMain from '../view/page-main.js';
+import NoData from '../view/no-data.js';
+import TripPresenter from './trip-presenter.js';
+import { render, updateItem } from '../render.js';
+
+export default class BoardPresenter {
+  #tripContainer = null;
+
+  #HeaderComponent = new PageHeader();
+  #MainComponent = new PageMain();
+  #TripList =  new TripList();
+  #sortComponent = new MainSort();
+  #noTripsComponent = new NoData();
+
+  #trips = [];
+  #tripPresenter = new Map();
+
+  constructor(tripContainer) {
+    this.#tripContainer = tripContainer;
+  }
+
+  init = (trips) => {
+    this.#trips = [...trips];
+
+    this.#renderBoard(this.#tripContainer);
+  }
+
+  #handleModeChange = () => {
+    this.#tripPresenter.forEach((presenter) => presenter.resetView());
+  }
+
+  #handleTripChange = (updatedTrip) => {
+    this.#trips = updateItem(this.#trips, updatedTrip);
+    this.#tripPresenter.get(updatedTrip.id).init(updatedTrip);
+  }
+
+
+  #clearTaskList = () => {
+    this.#tripPresenter.forEach((presenter) => presenter.destroy());
+    this.#tripPresenter.clear();
+  }
+
+  #renderTrip = (listElement, trip) => {
+    const tripPresenter = new TripPresenter(listElement, this.#handleTripChange, this.#handleModeChange);
+    tripPresenter.init(trip);
+    this.#tripPresenter.set(trip.id, tripPresenter);
+  }
+
+  #renderInfo = (container) => {
+    render(container, new InfoMain(this.#trips));
+    const tripCost = this.#trips.reduce((accumulator, trip) => accumulator + trip.price, 0);
+    render(container, new TripCost(tripCost));
```

Comment:
тога можно отказаться от локальной переменной - это всегда хорошо
```suggestion
    render(container, new TripCost(getTotalCost(this.#trips));
```